source/Nuke.Common/NukeBuild.cs

```diff
@@ -65,14 +65,23 @@ namespace Nuke.Common
     [SerializeBuildServerState]
     public abstract partial class NukeBuild : INukeBuild
     {
+        /// <summary>
+        /// Gets or sets the build exit code.
+        /// When set to <value>null</value> (default), <see cref="Execute{T}"/> will return a <em>0</em> exit code on build success; or a <em>-1</em> exit code on build failure.
+        /// When set to a non-null value, <see cref="Execute{T}"/> will return the value of <see cref="ExitCode"/>.
+        /// </summary>
+        protected static int? ExitCode { get; set; }
+        
         /// <summary>
         /// Executes the build. The provided expression defines the <em>default</em> target that is invoked,
         /// if no targets have been specified via command-line arguments.
         /// </summary>
         protected static int Execute<T>(params Expression<Func<T, Target>>[] defaultTargetExpressions)
             where T : NukeBuild
         {
-            return BuildManager.Execute(defaultTargetExpressions);
+            var buildExitCode = BuildManager.Execute(defaultTargetExpressions);
+            
+            return ExitCode ?? buildExitCode;
```

Comment:
When `ExitCode` is non-static, this logic should be moved to `BuildManager` .... This method here is just for delegation.

Reply:
Absolutely, done!